src/agent/onefuzz-agent/src/tasks/coverage/generic.rs

```diff
@@ -339,11 +344,24 @@ impl<'a> TaskContext<'a> {
 
     pub async fn save_and_sync_coverage(&self) -> Result<()> {
         let path = self.config.coverage.local_path.join(COVERAGE_FILE);
-        let text = serde_json::to_string(&self.coverage).context("serializing coverage to JSON")?;
-
+        let text = serde_json::to_string(&self.coverage).context("serializing block coverage")?;
         fs::write(&path, &text)
             .await
             .with_context(|| format!("writing coverage to {}", path.display()))?;
+
+        let path = self.config.coverage.local_path.join(SOURCE_COVERAGE_FILE);
+        let src_coverage = {
+            let mut debuginfo = self
+                .debuginfo
+                .lock()
+                .map_err(|e| anyhow::format_err!("{}", e))?;
+            self.coverage.source_coverage(&mut *debuginfo)?
```

Comment:
This method potentially does a lot of blocking IO. We should call it via `spawn_blocking()`.